narwhal/worker/src/handlers.rs

```diff
@@ -27,25 +28,35 @@ use types::{
 
 use sui_metrics::monitored_future;
 
+use crate::TxValidator;
+
 #[cfg(test)]
 #[path = "tests/handlers_tests.rs"]
 pub mod handlers_tests;
 
 /// Defines how the network receiver handles incoming workers messages.
 #[derive(Clone)]
-pub struct WorkerReceiverHandler {
+pub struct WorkerReceiverHandler<V> {
     pub id: WorkerId,
     pub tx_others_batch: Sender<WorkerOthersBatchMessage>,
     pub store: Store<BatchDigest, Batch>,
+    pub validator: V,
 }
 
 #[async_trait]
-impl WorkerToWorker for WorkerReceiverHandler {
+impl<V: TxValidator> WorkerToWorker for WorkerReceiverHandler<V> {
     async fn report_batch(
         &self,
         request: anemo::Request<WorkerBatchMessage>,
     ) -> Result<anemo::Response<()>, anemo::rpc::Status> {
         let message = request.into_body();
+        if self.validator.validate_batch(&message.batch).is_err() {
```

Comment:
There is one more point where we'll need to do batch verification and this is when we "synchronize" missing batches. The process is done here:

https://github.com/MystenLabs/sui/blob/f9698507a4cfe274676e940c863b4ecd518ff39e/narwhal/worker/src/handlers.rs#L242

where the request/response pattern is used and the batch is returned directly from the other party

Reply:
Yes, ofc.